File: mcm-backend/mcm-core/src/main/java/at/ac/tuwien/model/change/management/core/model/Configuration.java
package at.ac.tuwien.model.change.management.core.model;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;

@Setter
@Getter
@EqualsAndHashCode
public class Configuration {
    private String name;
    private ConfigurationVersion version; // hash of the git commit
    private Set<Model> models = new HashSet<>();

    public String getVersionHash() {
        return version != null ? version.hash() : null;
    }

    public String getVersionName() {
        return version != null ? version.name() : null;
    }

    public String getVersionCustomName() {
        return version != null ? version.customName() : null;
    }
}

File: mcm-frontend/src/datamodel/Configuration.ts
import {Model} from "@/datamodel/Model.ts";

export type Configuration = {
    id: string
    name: string
    path: string
    models: Model[]
}

export const configurations_data = [
    {
        id: '1',
        name: 'Configuration 1',
        path: 'path/to/configuration1',
        models: [
            {
                id: 1,
                name: 'Model 1'
            },
            {
                id: 2,
                name: 'Model 2'
            }
        ],
    },
]